### PR TITLE
[Newgrounds] Improve extractor and fix broken playlist extractor

### PR DESCRIPTION
**Pull request by u-spec-png**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fix broken playlist extractor, fix broken test and some improvements.

### CODE REVIEW

File: yt_dlp/extractor/newgrounds.py [hunk 1]
```diff
@@ -124,14 +131,19 @@ def _real_extract(self, url):
                  r'(?:Author|Writer)\s*<a[^>]+>([^<]+)'), webpage, 'uploader',
                 fatal=False)
 
-        timestamp = unified_timestamp(self._html_search_regex(
+        timestamp = self._html_search_regex(
             (r'<dt>\s*Uploaded\s*</dt>\s*<dd>([^<]+</dd>\s*<dd>[^<]+)',
              r'<dt>\s*Uploaded\s*</dt>\s*<dd>([^<]+)'), webpage, 'timestamp',
-            default=None))
+            default=None)
+        upload_date = unified_strdate(timestamp)
```
Comment (pukkandan) [inline yt_dlp/extractor/newgrounds.py:138]: will be automatically set from timestamp

File: yt_dlp/extractor/newgrounds.py [hunk 2]
```diff
@@ -140,17 +152,17 @@ def _real_extract(self, url):
 
         if '<dd>Song' in webpage:
             formats[0]['vcodec'] = 'none'
-
```
Comment (pukkandan) [inline yt_dlp/extractor/newgrounds.py:143]: Move sort_formats here

**Review by pukkandan — COMMENTED** [review]
Havent tested yet, but the code looks good

File: yt_dlp/extractor/newgrounds.py (hunk 2 above)
Comment (u-spec-png) [reply yt_dlp/extractor/newgrounds.py:143]: done

File: yt_dlp/extractor/newgrounds.py [hunk 3]
```diff
@@ -73,38 +102,16 @@ def _real_extract(self, url):
...
-            url_video_data = 'https://www.newgrounds.com/portal/video/%s' % video_id
-            headers = {
+            json_video = self._download_json('https://www.newgrounds.com/portal/video/' + media_id, media_id, headers={
                 'Accept': 'application/json',
                 'Referer': url,
                 'X-Requested-With': 'XMLHttpRequest'
-            }
-            json_video = self._download_json(url_video_data, video_id, headers=headers, fatal=False)
+            }, fatal=False)
             if not json_video:
```
Comment (Ashish0804) [inline yt_dlp/extractor/newgrounds.py:110]: This is not a suggestion or comment but a question

Why not use `fatal=True`?
```py
            if fatal:
                raise ExtractorError(errmsg, sys.exc_info()[2], cause=err)
```
this how the `fatal` argument is used.

Comment (pukkandan) [reply yt_dlp/extractor/newgrounds.py:110]: You r right. This code was written in 2014 and last updated seriously in 2017. I assume the fatal keyword didnt exist back then. But we should be using it now

Comment (u-spec-png) [reply yt_dlp/extractor/newgrounds.py:110]: I don't know it was like that i forgot to change it.

Comment (Ashish0804) [reply yt_dlp/extractor/newgrounds.py]: Here's the func
```py
def _download_json(
            self, url_or_request, video_id, note='Downloading JSON metadata',
            errnote='Unable to download JSON metadata', transform_source=None,
            fatal=True, encoding=None, data=None, headers={}, query={},
            expected_status=None)
```
So you can add the error note in `errnote`.

File: yt_dlp/extractor/newgrounds.py [hunk 4]
```diff
@@ -4,54 +4,81 @@
 
 from .common import InfoExtractor
 from ..utils import (
-    ExtractorError,
     extract_attributes,
     int_or_none,
     parse_duration,
     parse_filesize,
+    parse_count,
```
Comment (Ashish0804) [inline yt_dlp/extractor/newgrounds.py:11]: this should be at line 9

Comment (u-spec-png) [reply yt_dlp/extractor/newgrounds.py:11]: Ok, but why?

File: yt_dlp/extractor/newgrounds.py [hunk 5]
```diff
@@ -4,54 +4,81 @@
...
         'info_dict': {
             'id': '549479',
             'ext': 'mp3',
-            'title': 'Burn7 - B7 - BusMode',
+            'title': 'B7 - BusMode',
             'uploader': 'Burn7',
             'timestamp': 1378878540,
             'upload_date': '20130911',
             'duration': 143,
+            'description': 'Sonic blahblahblah I\'m late again SEPTEMBER 11TH: Busmode-The story of ants on a log. ',
```
Comment (Ashish0804) [inline yt_dlp/extractor/newgrounds.py:29]: nothing wrong with using `\'` but you can also use md5.
paste the text here `https://www.md5hashgenerator.com/` and then add it as 
`'description': 'md5:<md5hash>',`

File: yt_dlp/extractor/newgrounds.py (hunk 4 above)
Comment (Ashish0804) [reply yt_dlp/extractor/newgrounds.py:11]: `c` comes before `d`
tbh idk if there's guideline to sort the imports but it looks clean and not out of place.

Comment (u-spec-png) [reply yt_dlp/extractor/newgrounds.py]: Thanks

File: yt_dlp/extractor/newgrounds.py (hunk 3 above)
Comment (u-spec-png) [reply yt_dlp/extractor/newgrounds.py]: I mean there is nothing wrong with default error note there is no need for custom ```errnote```.

**Pull request merged**